lib/Models/MapboxVectorTileCatalogItem.ts

```diff
@@ -0,0 +1,152 @@
+import ImageryProvider from "terriajs-cesium/Source/Scene/ImageryProvider";
+import isDefined from "../Core/isDefined";
+import MapboxVectorTileImageryProvider from "../Map/MapboxVectorTileImageryProvider";
+import AsyncMappableMixin from "../ModelMixins/AsyncMappableMixin";
+import CatalogMemberMixin from "../ModelMixins/CatalogMemberMixin";
+import UrlMixin from "../ModelMixins/UrlMixin";
+import LegendTraits, { LegendItemTraits } from "../Traits/LegendTraits";
+import MapboxVectorTileCatalogItemTraits from "../Traits/MapboxVectorTileCatalogItemTraits";
+import CreateModel from "./CreateModel";
+import createStratumInstance from "./createStratumInstance";
+import LoadableStratum from "./LoadableStratum";
+import { BaseModel } from "./Model";
+import StratumOrder from "./StratumOrder";
+
+class MapboxVectorTileLoadableStratum extends LoadableStratum(
+  MapboxVectorTileCatalogItemTraits
+) {
+  static stratumName = "MapboxVectorTileLoadable";
+
+  constructor(readonly item: MapboxVectorTileCatalogItem) {
+    super();
+  }
+
+  duplicateLoadableStratum(newModel: BaseModel): this {
+    return new MapboxVectorTileLoadableStratum(
+      newModel as MapboxVectorTileCatalogItem
+    ) as this;
+  }
+
+  static async load(item: MapboxVectorTileCatalogItem) {
+    return new MapboxVectorTileLoadableStratum(item);
+  }
+
+  @computed get legends() {
+    return [
+      createStratumInstance(LegendTraits, {
+        items: [
+          createStratumInstance(LegendItemTraits, {
+            color: this.item.fillColor,
+            outlineColor: this.item.lineColor,
+            title: this.item.name
+          })
+        ]
+      })
+    ];
+  }
+}
+
+StratumOrder.addLoadStratum(MapboxVectorTileLoadableStratum.stratumName);
+
+class MapboxVectorTileCatalogItem extends AsyncMappableMixin(
+  UrlMixin(CatalogMemberMixin(CreateModel(MapboxVectorTileCatalogItemTraits)))
+) {
+  @observable
+  private imageryProvider: ImageryProvider | undefined;
+
+  public readonly forceProxy = true;
+
+  static readonly type = "mvt";
+
+  get type() {
+    return MapboxVectorTileCatalogItem.type;
+  }
+
+  get typeName() {
+    return i18next.t("models.mapboxVectorTile.name");
+  }
+  readonly canZoomTo = true;
+
+  async forceLoadMetadata() {
+    const stratum = await MapboxVectorTileLoadableStratum.load(this);
+    runInAction(() => {
+      this.strata.set(MapboxVectorTileLoadableStratum.stratumName, stratum);
+    });
+  }
+
+  protected async forceLoadMapItems(): Promise<void> {
+    await this.loadMetadata();
+    this.updateImageryProvider();
+  }
+
+  private updateImageryProvider() {
+    autorun(() => {
+      if (
+        !isDefined(this.url) ||
+        !isDefined(this.layer) ||
+        !isDefined(this.rectangle)
+      ) {
+        return;
+      }
+
+      this.imageryProvider = new MapboxVectorTileImageryProvider({
+        url: this.url,
+        layerName: this.layer,
+        styleFunc: action(() => ({
```

Comment:
You could avoid this action using a closure. e.g:

```js
styleFunc: ((style) => () => style)({
  fillStyle: this.fillColor, 
  strokeStyle: this.lineColor, 
  lineJoin: "miter", 
  lineWidth: 1
})
```

or:

```js
styleFunc: (({fillColor, lineColor}) => () => ({
  fillStyle: fillColor, 
  strokeStyle: lineColor, 
  lineJoin: "miter", 
  lineWidth: 1
}))({
  fillColor: this.fillColor, 
  lineColor: this.lineColor
})
```
This will evaluate the observables/computed early and the resultant `styleFunc` will no longer use any tracked variables

Reply:
Good point again!